patch.py

```
@@ -0,0 +1,162 @@
+reserve_mandatory = """
+    pub fn reserve_mandatory(&mut self, additional: usize) {
+        for i in [0, 4] { if let Some(col) = self.columns.get_mut(i) { col.reserve(additional); } }
+    }
+    pub fn reserve_string_buf(&mut self, additional: usize) {
+        self.string_buf.reserve(additional);
+    }
+"""
+patch("crates/ffwd-arrow/src/columnar/builder.rs", "    pub fn begin_batch(&mut self) {", reserve_mandatory + "    pub fn begin_batch(&mut self) {")
+
+# 4. wire.rs
+with open(base + "crates/ffwd-io/src/otlp_receiver/projection/wire.rs", 'r') as f:
```

Ruff (0.15.11)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)